Cake.Recipe/Content/localbuild.cake

```diff
@@ -47,33 +53,60 @@ public class LocalBuildRepositoryInfo : IRepositoryInfo
 {
     public LocalBuildRepositoryInfo(ICakeContext context)
     {
-        context.Information("Getting current branch name...");
-        IEnumerable<string> redirectedStandardOutput;
-        IEnumerable<string> redirectedError;
+        try
+        {
+            context.Information("Testing to see if valid git repository...")
```

Comment:
You forgot a semi-colon here

Reply:
Dang it!  The amount of times that I do this!  Fixed...

Reply:
Which is exactly why I have the following in the `.git/hooks/pre-push` script for Cake.Recipe:

```
#!/bin/sh

# git pre-push hook

remote="$1"
url="$2"

z40=0000000000000000000000000000000000000000
while read local_ref local_sha remote_ref remote_sha
do
	if [ "$local_sha" = $z40 ]
	then
		# Handle delete
		:
	else
		if [ "$remote_sha" = $z40 ]
		then
			# New branch, examine all commits
			range="$local_sha"
		else
			# Update to existing branch, examine new commits
			range="$remote_sha..$local_sha"
		fi

		# Check for WIP commit
		commit=`git rev-list -n 1 --grep '^WIP' "$range"`
		if [ -n "$commit" ]
		then
			echo >&2 "Found WIP commit in $local_ref, not pushing"
			exit 1
		fi
	fi
done

store=$(git diff --name-only 2>/dev/null || git diff --name-only --cached)

if [ ! -z "$store" ]; then
  git stash --include-untracked 2>/dev/null
fi

pwsh -NoProfile ./build.ps1 # Or powershell.exe

exitCode="$?"

if [ ! -z "$store" ]; then
  git stash pop --index >/dev/null
fi

exit $exitCode
```

Reply:
@AdmiringWorm never really played with git hooks before, but going to take this one for a spin!  Thanks for showing me this!

Reply:
:+1:
btw, since you are using OSX, you could probably change `pwsh -NoProfile ./build.ps1 # Or powershell.exe` to `./build.sh` instead.